[P9] 8170-Pharo-image-freeze-on-Windows-after-some-hours-Idle

tesonep opened the pull request:
Fix #8170 for P9

tesonep committed:
Adding a new event that records changes in the display or resolutions.
This is needed to correctly handle Remote Desktop connection and Disconnection.

tesonep committed:
Adding a catch here because we need to ensure that the error does not destroy everything.
The problem is if we have the exception in a mutex, and the mutex is not liberated.

Ducasse commented on src/OSWindow-SDL2/SDL_SysWMEvent.class.st:
```diff
@@ -0,0 +1,122 @@
+	^ SDL_SYSWMEVENT
+]
+
+{ #category : #'as yet unclassified' }
```
as yet unclassified

Ducasse commented on src/OSWindow-Core/OSWindowMorphicEventHandler.class.st:
```diff
@@ -351,6 +351,14 @@ OSWindowMorphicEventHandler >> visitWindowResizeEvent: anEvent [
+	"We need to reset the render if the resolution changes. 
+	If we don't do this the image seems blocked but as it does not refresh the window"
+	
+	morphicWorld worldState worldRenderer window backendWindow renderer destroy.
```
May be we should open an issue because this line shows that there is something somehow wrong.

tesonep committed:
Spliting the initialization of constants. 
The Pharo 9 bootstrap is still building the image first in V3 bytecode and the number of literals is limited.
These methods can be inlined when V3 bytecode is left behind in the bootstrap.

tesonep committed:
Categorizing methods

tesonep committed:
Refactoring of the tasks to do in the change of resolution

tesonep commented:
We are testing it :D

Ducasse commented:
Ok we will get there. This is probably a canvas contents that got shifted.

VincentBlondeau commented:
@demarey Sure! You can then close this issue.
